### ISSUE TITLE
docker-env arg in minikube start is able to parse only first value for NO_PROXY from list of domains

### ISSUE THREAD

**Issue by spk83** [kubernetes/minikube#811]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Minikube version** (use `minikube version`): v0.12.2

**Environment**:
- **OS** :OS X 10.11.6
- **VM Driver** : xhyve
- **Docker version** (e.g. `docker -v`): Docker 1.12.2 build bb80604
- **Install tools**: brew and brew cask
- **Others**: docker-machine-driver-xhyve 0.3.0

**What happened**: When I pass list of domains to bypass in NO_PROXY env variable with docker-env arg, it only takes first value out of the list (that is comma separated). Once machine is up (after the error), if I ssh and cat the /var/lib/boot2docker/profile I can see that NO_PROXY isn't getting exported correctly. 

**What you expected to happen**:
In minikube start command, docker-env should be able to read NO_PROXY="list of domains(comma separated)" as key=value 

**How to reproduce it** (as minimally and precisely as possible):
```
minikube start \
  --vm-driver xhyve \
  --docker-env NO_PROXY=".local,localhost,.abc.com,.xyz.com,127.0.0.1"
```
It will throw error like below:
```
Starting local Kubernetes cluster...
E1108 16:21:58.454157   23069 start.go:92] Error starting host: Error creating host: Error creating machine: Error running provisioning: Something went wrong running an SSH command!
command : sudo /etc/init.d/docker start
err     : exit status 2
output  : /etc/init.d/docker: export: line 15: .abc.com: bad variable name

. Retrying.
E1108 16:21:58.455195   23069 start.go:98] Error starting host:  Error creating host: Error creating machine: Error running provisioning: Something went wrong running an SSH command!
command : sudo /etc/init.d/docker start
err     : exit status 2
output  : /etc/init.d/docker: export: line 15: .abc.com: bad variable name
```

And /var/lib/boot2docker/profile is created as below
```
> minikube ssh -- cat /var/lib/boot2docker/profile

EXTRA_ARGS='
--label provider=xhyve

'
CACERT=/var/lib/boot2docker/ca.pem
DOCKER_HOST='-H tcp://0.0.0.0:2376'
DOCKER_STORAGE=aufs
DOCKER_TLS=auto
SERVERKEY=/var/lib/boot2docker/server-key.pem
SERVERCERT=/var/lib/boot2docker/server.pem

export "NO_PROXY=.local"
export "localhost"
export ".abc.com"
export ".xyz.com"
export "127.0.0.1"

```

**Anything else do we need to know**: Dont face any issues when passing list of domains as no_proxy to docker-machine create command's --engine-env arg. Below is the /var/lib/boot2docker/profile from docker-machine. 
```
> docker-machine create \
     -d xhyve \
     --engine-env no_proxy=".local,localhost,.abc.com,.xyz.com,127.0.0.1" \
    default
....
....
> docker-machine ssh default -- cat /var/lib/boot2docker/profile

EXTRA_ARGS='
--label provider=xhyve

'
CACERT=/var/lib/boot2docker/ca.pem
DOCKER_HOST='-H tcp://0.0.0.0:2376'
DOCKER_STORAGE=aufs
DOCKER_TLS=auto
SERVERKEY=/var/lib/boot2docker/server-key.pem
SERVERCERT=/var/lib/boot2docker/server.pem

export "no_proxy=.local,localhost,.abc.com,.xyz.com,127.0.0.1"

```
Please let me know if any other info is needed. I will try to provide as quickly I can.

**Comment by r2d4** [comment]
Hey thanks for the detailed report.  It looks like this is happening because of the flag library we use.

Here is the issue upstream https://github.com/spf13/pflag/issues/81

We will need to upgrade spf13/pflag to include this merge
https://github.com/spf13/pflag/pull/82

**Comment by spk83** [comment]
thanks. would you like me to open a pull request for this?

**Comment by r2d4** [comment]
Yeah, that would be great.  Take a look at this issue if you need some more guidance https://github.com/kubernetes/kubernetes/issues/27454

You will also need to change dockerEnv to use StringArrayVar

```
-       startCmd.Flags().StringSliceVar(&dockerEnv, "docker-env", nil, "Environment variables to pass to the Docker daemon. (format: key=value)")
+       startCmd.Flags().StringArrayVar(&dockerEnv, "docker-env", nil, "Environment variables to pass to the Docker daemon. (format: key=value)")
```

Feel free to ping me if you have any questions

**Comment by r2d4** [comment]
Fixed by #812
